src/OrchardCore/OrchardCore.ReCaptcha.Core/ActionFilters/Detection/IPAddressRobotDetector.cs

```diff
@@ -52,7 +52,7 @@ public void FlagAsRobot()
 
         // This has race conditions, but it's ok.
         var faultyRequestCount = _memoryCache.GetOrCreate(ipAddressKey, fact => 0);
-        faultyRequestCount++;
-        _memoryCache.Set(ipAddressKey, faultyRequestCount);
+
+        _memoryCache.Set(ipAddressKey, ++faultyRequestCount);
```

Comment:
How about using `IDistributedCache` here instead, since this will be node-local, as you observed?

Reply:
I'll do this is a separate PR along with my other notes.